spec/models/order_spec.rb

```diff
@@ -0,0 +1,112 @@
+require "rails_helper"
+RSpec.describe Order, type: :model do
+  describe "associations" do
+    it { is_expected.to belong_to(:user) }
+    it { is_expected.to belong_to(:payment_method) }
+    it { is_expected.to belong_to(:address) }
+    it { is_expected.to have_many(:order_items).dependent(:destroy) }
+  end
+
+  describe "nested attributes" do
+    it { is_expected.to accept_nested_attributes_for(:order_items) }
+    it { is_expected.to accept_nested_attributes_for(:address) }
+    it { is_expected.to accept_nested_attributes_for(:payment_method) }
+  end
+
+  describe "enums"  do
+    it { is_expected.to define_enum_for(:status).with_values(failed: 0, succeeded: 1, confirming: 2, rejected: 3) }
+  end
+
+  describe "monetize" do
+    it { is_expected.to monetize(:total_invoice_cents).with_model_currency(:total_invoice_currency) }
+  end
+
+    describe ".with_status" do
+      let(:user) { create(:user) }
+      let(:payment_method) { create(:payment_method) }
+      let(:address) { create(:address, user: user) }
+      let(:order1) { create :order, user: user, status: :succeeded, created_at: Time.current}
+      let(:order2) { create :order, user: user, status: :failed, created_at: Time.current}
```

Comment:
let vs let! khác gì nhau

Reply:
let tạo giá trị khi nó được gọi trong khối it ; còn let! nó tạo giá trị luon trong khối before